lib/ddtrace/propagation/http_propagator.rb

```diff
@@ -11,26 +15,65 @@ module HTTPPropagator
     def self.inject!(context, env)
       # Prevent propagation from being attempted if context provided is nil.
       if context.nil?
-        Datadog::Tracer.log.debug('Cannot inject context into env to propagate over HTTP: context is nil.'.freeze)
+        ::Datadog::Tracer.log.debug('Cannot inject context into env to propagate over HTTP: context is nil.'.freeze)
         return
       end
 
-      env[HTTP_HEADER_TRACE_ID] = context.trace_id.to_s
-      env[HTTP_HEADER_PARENT_ID] = context.span_id.to_s
-      env[HTTP_HEADER_SAMPLING_PRIORITY] = context.sampling_priority.to_s
-      env[HTTP_HEADER_ORIGIN] = context.origin.to_s if context.origin
-      env.delete(HTTP_HEADER_SAMPLING_PRIORITY) unless context.sampling_priority
+      # Check if they want to propagate with B3 headers
+      if ::Datadog.configuration.propagation_inject_style.include?(PROPAGATION_STYLE_B3)
```

Comment:
A list of of `if`s usually is a code smell to me. Does this inject each kind of header if its in the list on the global settings object?

Perhaps there's a way to implement this so its a bit more dynamic (maybe it iterates over a list, uses a Hash, etc.) I think doing so would be worthwhile since it wouldn't be far fetched if someone else wanted to define their own propagation style, and add it the list of available styles.

Reply:
The RFC defines the value of both settings as a list of strings. At this time it probably wouldn't be possible for someone to define their own propagation style, but something to consider for the next iteration of this.

Reply:
Couldn't they just add their own style to the list of strings?

Reply:
ah, yeah, I guess, if we also store a reference to all of these styles in a hash on configuration as well.

that feels a little like over-engineering for now?

Reply:
not sure what custom HTTP propagation logic people would want?

Reply:
I can say that I know at least one user who does custom HTTP propagation. But general intent is that we make the code cleaner by decoupling things, and permitting new standards to be easily added (as they do seem to pop up with all the cross compatibility we have.)

If this were one, or just two different styles we had to support, then I'd agree it might be over-engineering. But I think there's now a demonstrated pattern, and just a bit more work would yield a lot of benefit.

Reply:
I am reluctant to expose the ability for people to add their own propagators at this time, but we can definitely refactor this code to be more clean and better setup for that possibility in the future.

Reply:
Yeah, as stated on the original review, building out a full-fledged public API is out of scope of this PR. But we can do some changes here to at least not repeat a lot of `if` statements which would clean up the code a bit.

Reply:
let me know what you think of new flow